xchange-stream-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingMarketDataService.java

```diff
@@ -30,16 +31,32 @@ public KrakenStreamingMarketDataService(KrakenStreamingService service) {
     this.service = service;
   }
 
-  @Override
-  public Observable<OrderBook> getOrderBook(CurrencyPair currencyPair, Object... args) {
-    String channelName = getChannelName(KrakenSubscriptionName.book, currencyPair);
-    OrderBook orderBook = new OrderBook(null, Lists.newArrayList(), Lists.newArrayList());
-    int depth = parseOrderBookSize(args);
-    return subscribe(channelName, MIN_DATA_ARRAY_SIZE, depth)
-        .map(
-            arrayNode ->
-                KrakenStreamingAdapters.adaptOrderbookMessage(orderBook, currencyPair, arrayNode));
-  }
+    @Override
+    public Observable<OrderBook> getOrderBook(CurrencyPair currencyPair, Object... args) {
+      String channelName = getChannelName(KrakenSubscriptionName.book, currencyPair);
+      TreeSet<LimitOrder> bids = Sets.newTreeSet();
+      TreeSet<LimitOrder> asks = Sets.newTreeSet();
+      Object lock = new Object();
+      int depth = KrakenStreamingService.parseOrderBookSize(args);
+      return subscribe(channelName, MIN_DATA_ARRAY_SIZE, args).map(arrayNode -> {
```

Comment:
is there not a way we can just create a new orderbook from scratch when needed (bids/asks) rather then use synchronization? seems super heavy handed and more dangerous for devs who may not be as familiar with this